src/analyze/analyze.c

```diff
@@ -1832,6 +2010,50 @@ static int do_security(int argc, char *argv[], void *userdata) {
         return analyze_security(bus, strv_skip(argv, 1), 0);
 }
 
+static int parse_arg_color_style(DotArgColorStyle color_or_style, const char *arg_dot_color_style) {
+        char delim[] = ":";
+        const char *p;
+        int r;
+
+        p = arg_dot_color_style;
+        _cleanup_free_ char *word = NULL;
+
+        r = extract_first_word(&p, &word, delim, 0);
+        if (r == 0)
+                return r;
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0) {
+                log_error("Invalid syntax, ignoring: %s", arg_dot_color_style);
+                return r;
```

Comment:
`return log_error_errno(r, …`

Also, you are not ignoring the error here, don#t say you do

Reply:
done

Reply:
return and log_error lines still need to be folded into one